src/server/auth/server/oidc.go

```diff
@@ -118,6 +63,31 @@ func CryptoString(n int) string {
 	return base64.RawURLEncoding.EncodeToString(b)
 }
 
+// validateOIDC validates an OIDC configuration before it's stored in etcd.
+func validateOIDCConfig(config *auth.AuthConfig) error {
+	if _, err := url.Parse(config.Issuer); err != nil {
+		return errors.Wrapf(err, "OIDC issuer must be a valid URL")
+	}
+
+	// this does a request to <issuer>/.well-known/openid-configuration to see if it works
+	ctx := context.Background()
+	_, err := oidcProvider(ctx, config)
```

Comment:
I know this is in the OIDC library, but do you know offhand if this potentially blocks (or leaks the goroutine) if the issuer is unavailable?

Reply:
Updated this to accept the incoming ctx from `SetConfiguration` so it can be cancelled. AFAICT there's no risk of a goroutine leaking from this, it looks like `NewProvider` fetches the well-known URI during the constructor using the provider context, and the JWKS library only makes a request if you call Verify for an un-cached kid (https://github.com/coreos/go-oidc/blob/v3.0.0/oidc/jwks.go#L118)